### PR TITLE
Update map when crossing is clicked

### PR DESCRIPTION
**Pull request by briaguya**
I updated the map component so when a user clicks a crossing the map centers on it and the icon changes to be a selected icon. In order to do this I utilized the mapbox filter functionality, and added a layer for selected crossings. During the styling story, we should implement more layers so we can have different icons for selected crossings based on their status.

### CODE REVIEW

**Comment by mateoclarke** [comment]
Make sure we add this to Asana?

> During the styling story, we should implement more layers so we can have different icons for selected crossings based on their status.

File: frontend/src/Map/CrossingMap.js [hunk 1]
```diff
@@ -8,11 +8,19 @@ import ReactMapboxGl, { Layer, Feature } from "react-mapbox-gl";
 const Map = ReactMapboxGl({ accessToken: null });
 
 class CrossingMap extends React.Component {
+  state = {
+    selectedCrossingId: -1,
```
Comment (mateoclarke) [inline frontend/src/Map/CrossingMap.js]: Does this get the last ID? Can you explain this?

File: frontend/src/Map/CrossingMap.js [hunk 2]
```diff
@@ -36,29 +49,39 @@ class CrossingMap extends React.Component {
...
+        center={this.state.center}>
         <Layer
           type="symbol"
           id="closedCrossings"
           layout={{ 'icon-image': 'cross-15', 'icon-allow-overlap': true }}
+          layerOptions={{"filter":
+            [
+              "all",
+              ["!=", "id", this.state.selectedCrossingId],
+              ["==", "crossingStatus", 2]
```
Comment (mateoclarke) [inline frontend/src/Map/CrossingMap.js]: I don't understand what these layerOptions do and it wasn't very specific in the docs.
https://github.com/alex3165/react-mapbox-gl/blob/master/docs/API.md

I see we use a fixed number here, 2 and 1 somewhere else. What are these magic numbers?

Comment (briaguya) [reply frontend/src/Map/CrossingMap.js]: The LayerOptions object gets passed through the react-mapbox-gl wrapper into the mapbox map. I'm passing in the [filter](https://www.mapbox.com/mapbox-gl-js/style-spec/#types-filter).

**Pull request merged**